src/Rest/Routes/FormSubmitCaptchaRoute.php

```diff
@@ -0,0 +1,168 @@
+ *
+ * @package EightshiftForms\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Rest\Routes;
+
+use EightshiftForms\Helpers\Components;
+use EightshiftForms\Hooks\Variables;
+use EightshiftForms\Settings\SettingsHelper;
+use EightshiftForms\Labels\LabelsInterface;
+use EightshiftForms\Validation\SettingsCaptcha;
+
+/**
+ * Class FormSubmitCaptchaRoute
+ */
+class FormSubmitCaptchaRoute extends AbstractBaseRoute
+{
+	/**
+	 * Use general helper trait.
+	 */
+	use SettingsHelper;
+
+	/**
+	 * Instance variable of LabelsInterface data.
+	 *
+	 * @var LabelsInterface
+	 */
+	protected $labels;
+
+	/**
+	 * Create a new instance that injects classes
+	 *
+	 * @param LabelsInterface $labels Inject LabelsInterface which holds labels data.
+	 */
+	public function __construct(LabelsInterface $labels)
+	{
+		$this->labels = $labels;
+	}
+
+	/**
+	 * Get the base url of the route
+	 *
+	 * @return string The base URL for route you are adding.
+	 */
+	protected function getRouteName(): string
+	{
+		return '/form-submit-captcha';
+	}
+
+	/**
+	 * Get callback arguments array
+	 *
+	 * @return array<string, mixed> Either an array of options for the endpoint, or an array of arrays for multiple methods.
+	 */
+	protected function getCallbackArguments(): array
+	{
+		return [
+			'methods' => $this->getMethods(),
+			'callback' => [$this, 'routeCallback'],
+			'permission_callback' => [$this, 'permissionCallback'],
+		];
+	}
+
+	/**
+	 * Method that returns rest response
+	 *
+	 * @param \WP_REST_Request $request Data got from endpoint url.
+	 *
+	 * @return \WP_REST_Response|mixed If response generated an error, WP_Error, if response
+	 *                                is already an instance, WP_HTTP_Response, otherwise
+	 *                                returns a new WP_REST_Response instance.
+	 */
+	public function routeCallback(\WP_REST_Request $request)
+	{
+		$params = json_decode($request->get_body(), true);
+
+		$token = $params['token'] ?? '';
+		$formId = $params['formId'] ?? '';
+
+		if (!$token || !$formId) {
+			return \rest_ensure_response([
+				'status' => 'error',
+				'code' => 400,
+				'message' => $this->labels->getLabel('mailerSuccessNoSend', $formId),
+			]);
+		}
+
+		$secretKey = !empty(Variables::getGoogleReCaptchaSecretKey()) ? Variables::getGoogleReCaptchaSecretKey() : $this->getOptionValue(SettingsCaptcha::SETTINGS_CAPTCHA_SECRET_KEY);
+
+		$response = \wp_remote_post(
+			"https://www.google.com/recaptcha/api/siteverify",
+			[
+				'body' => [
+					'secret' => $secretKey,
+					'response' => $token,
+				],
+			]
+		);
+
+		// Generic error msg from WP.
+		if (is_wp_error($response)) {
+			return [
+				'status' => 'error',
+				'code' => 400,
+				'message' => $this->labels->getLabel('submitWpError', $formId),
+			];
+		}
+
+		// Get body from the response.
+		$responseBody = json_decode(\wp_remote_retrieve_body($response), true);
```

Comment:
Same as up - this will throw warnings if the response body is for any reason malformed or otherwise not valid JSON.

Reply:
nice catch tnx